watson/cli.py

```diff
@@ -1242,28 +1251,40 @@ def edit(watson, confirm_new_project, confirm_new_tag, id):
     datetime_format = '{} {}'.format(date_format, time_format)
     local_tz = tz.tzlocal()
 
-    if id:
-        frame = get_frame_from_argument(watson, id)
-        id = frame.id
+    if day:
+        # Editing all the frames of today
+        frames = get_frames_for_today(watson)
+    elif week:
+        # Editing all the frames of the week
+        frames = get_frames_for_week(watson)
+    elif id:
+        # Editing a single frame by id
+        frames = [get_frame_from_argument(watson, id)]
+        id = frames[0].id
     elif watson.is_started:
-        frame = Frame(watson.current['start'], None, watson.current['project'],
-                      None, watson.current['tags'])
+        # Frame is started and doesn't have a stop time yet
+        frames = [Frame(watson.current['start'], None,
+                        watson.current['project'], None,
+                        watson.current['tags'])]
     elif watson.frames:
-        frame = watson.frames[-1]
-        id = frame.id
+        # Id is not provided, edit the latest frame
+        frames = [watson.frames[-1]]
+        id = frames[0].id
     else:
         raise click.ClickException(
             style('error', "No frames recorded yet. It's time to create your "
                            "first one!"))
-
-    data = {
-        'start': frame.start.format(datetime_format),
-        'project': frame.project,
-        'tags': frame.tags,
-    }
-
-    if id:
-        data['stop'] = frame.stop.format(datetime_format)
+    data = [
+        {
+            'id': frame.id,
```

Comment:
This will break if we edit the current frame that has no id yet?

Reply:
Thank you for your review @jmaupetit. No this will not break as the id will have a value of `null` as follows:

```
[
    {
        "id": null,
        "project": "NoId",
        "start": "2020-07-16 21:02:09",
        "stop": "2020-07-16 21:02:12",
        "tags": [
            "Frame with no id yet"
        ]
    }
]
```

in this implementation, `id`s are immutable, meaning even if the user tries to change them they will be reverted back to original values. In this case `null`.